### PR TITLE
feat: restricted conference calling

### PR DESCRIPTION
**Pull request by alexandreferris**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

[AR-2535 - Calls restriction on non paying members](https://wearezeta.atlassian.net/browse/AR-2535)

This is the EPIC PR containing logic and tests (can see correct PRs below)

### Issues

Users that aren't in a paying team were able to start a conference/group call.

### Causes (Optional)

We weren't checking for this feature flag.

### Solutions

Correctly listen and verify the feature flag value when starting a conference/group call.

### Dependencies (Optional)

This is the epic PR

Needs releases with:

- [X] #1062 
- [X] #1064 

#### How to Test

It is not possible to test it yet, tests will be possible when Kalium side is merged and will be used in AR.

### CODE REVIEW

**Review by yamilmedina — APPROVED** [review]
Great job !  🎸

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/UserConfigRepository.kt [hunk 1]
```diff
@@ -16,6 +16,8 @@ interface UserConfigRepository {
     fun getClassifiedDomainsStatus(): Flow<Either<StorageFailure, ClassifiedDomainsStatus>>
     fun isMLSEnabled(): Either<StorageFailure, Boolean>
     fun setMLSEnabled(enabled: Boolean): Either<StorageFailure, Unit>
+    fun setConferenceCallingEnabled(enabled: Boolean): Either<StorageFailure, Unit>
+    fun isConferenceCallingEnabledFlow(): Flow<Either<StorageFailure, Boolean>>
```
Comment (ohassine) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/UserConfigRepository.kt:20]: Do we need to observe the flag ? checking before starting a call isn't enough ?

Comment (alexandreferris) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/UserConfigRepository.kt]: Nevermind. After actually having a good look of the code in junction with AR again it does seem more logical not having the observe/flow from the useCase and rather as you mentioned, returning it directly as a solely Boolean.

**Review by ohassine — APPROVED** [review]
well done 👌

**Pull request merged**